ui/src/app/components/demographic-survey-v2-validation-message.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import * as fp from 'lodash/fp';
+
+import { withProfileErrorModal } from 'app/components/with-error-modal';
+
+import { possiblePreferNotToAnswerErrors } from './demographic-survey-v2';
+import { WithSpinnerOverlayProps } from './with-spinner-overlay';
+
+interface DemographicSurveyValidationMessageProps
+  extends WithSpinnerOverlayProps {
+  errors: any;
+  isAccountCreation: boolean;
+  captcha?: boolean;
+  changed?: boolean;
+}
+
+const showPreferNotToAnswerMessage = (errors) =>
+  Object.keys(errors).some((error) =>
+    possiblePreferNotToAnswerErrors.includes(error)
+  );
+
+export const DemographicSurveyValidationMessage = fp.flow(
+  withProfileErrorModal
+)((props: DemographicSurveyValidationMessageProps) => {
+  const { captcha, changed, errors, isAccountCreation } = props;
+  return (
+    (errors || (isAccountCreation && !captcha)) && (
```

Comment:
I don't think this covers the `!isAccountCreation && !changed` case

Reply:
Good catch. Should be updated now.